packages/driver/src/cross-origin/origin_fn.ts

```diff
@@ -66,6 +61,38 @@ const rehydrateRunnable = (data: serializedRunnable): Runnable|Test => {
   return runnable
 }
 
+const getCallbackFn = async (fn: string, file?: string) => {
+  if (
+    // @ts-ignore
+    !Cypress.config('experimentalOriginDependencies')
+    || !fn.includes('Cypress.require')
+  ) {
+    return fn
+  }
+
+  const callbackName = '__cypressCallback'
+  const response = await fetch('/__cypress/process-origin-callback', {
```

Comment:
Ok, to ensure my understanding let me describe what i think is happening. 

1. We detect if `Cypress.require` exists in the callback, if it doesn't we just eval fn as is
2. If it does exist, we send it to the server
3. The server webpacks the fn to bundle in all the deps, then returns the packed bundle
4. We then eval the fn like normal.

Reply:
Yep, that's right 💯

Reply:
I found this explainer super helpful. Can we add a comment like this to `getCallbackFn`?

Reply:
Added in https://github.com/cypress-io/cypress/pull/25626/commits/b5270d5fad626668014d648d7199674cac7e3d0c